bin/miden-tx-prover/src/server/mod.rs

```diff
@@ -0,0 +1,72 @@
+use generated::api::api_server;
+use miden_objects::transaction::TransactionWitness;
+use miden_tx::{
+    utils::{Deserializable, Serializable},
+    LocalTransactionProver, TransactionProver,
+};
+use tokio::net::TcpListener;
+use tonic::{Request, Response, Status};
+use tracing::info;
+use winter_maybe_async::maybe_await;
+
+use crate::{ProveTransactionRequest, ProveTransactionResponse};
+
+pub mod generated;
+
+pub struct Rpc {
+    pub api_service: api_server::ApiServer<RpcApi>,
+    pub listener: TcpListener,
+}
+
+impl Rpc {
+    pub fn new(listener: TcpListener) -> Self {
+        let api_service = api_server::ApiServer::new(RpcApi);
+        Self { listener, api_service }
+    }
+}
+
+#[derive(Clone)]
+pub struct RpcApi;
+
+#[cfg(not(feature = "async"))]
+#[tonic::async_trait]
+impl api_server::Api for RpcApi {
+    async fn prove_transaction(
+        &self,
+        request: Request<ProveTransactionRequest>,
+    ) -> Result<Response<ProveTransactionResponse>, tonic::Status> {
+        info!("Received request to prove transaction");
+        let prover = LocalTransactionProver::default();
```

Comment:
Should we make `prover` an instance variable of `RpcApi`?

Reply:
That version will require changes in the `LocalTransactionProver`, since internally it uses `Rc` and cannot be shared between threads safely.

Reply:
I guess do we need to share it between threads? We should be processing only 1 request at a time and so may be able to put it behind a mutex or something like that.